### DISCUSSION TITLE
Is there a way to save a variable for use across multiple .http files?

### DISCUSSION THREAD

**Discussion by akiander** [Huachao/vscode-restclient#1241] [Q&A]
I know that file variables work within a single .http file. Is there a way to issue a request and then save a value from the response to a variable that can be used in multiple other .http files?

**Comment by willnationsdev** [comment]
Yes. See the README [here](https://github.com/Huachao/vscode-restclient?tab=readme-ov-file#request-variables) about "request variables".

Basically, if you use `@name` to assign a name to a request, then you can reference the response of that latest execution of that request to populate a variable.

**Reply by MominRaza** [reply]
no, request variables we can only use in single file

**Comment by MominRaza** [comment]
Related update: https://github.com/Huachao/vscode-restclient/pull/1422

This discussion is addressed by the new `@set` directive, which lets you capture response values (`response.headers.*` / `response.body.*`) into shared runtime variables and reuse them across `.http` files.
